pyuvdata/uvdata/uvdata.py

```diff
@@ -8530,68 +8526,80 @@ def _select_metadata(
-                ]
-                self.antenna_names = [self.antenna_names[ind] for ind in inds_to_keep]
-                self.antenna_numbers = self.antenna_numbers[inds_to_keep]
-                self.antenna_positions = self.antenna_positions[inds_to_keep, :]
-                if self.antenna_diameters is not None:
-                    self.antenna_diameters = self.antenna_diameters[inds_to_keep]
-                self.Nants_telescope = int(len(ants_to_keep))
-
-        if freq_inds is not None:
-            self.Nfreqs = len(freq_inds)
-            if self.future_array_shapes:
-                self.freq_array = self.freq_array[freq_inds]
-            else:
-                self.freq_array = self.freq_array[:, freq_inds]
-            if self.flex_spw or self.future_array_shapes:
-                self.channel_width = self.channel_width[freq_inds]
-            if self.flex_spw:
-                self.flex_spw_id_array = self.flex_spw_id_array[freq_inds]
-                # Use the spw ID array to check and see which SPWs are left
-                spw_screen = np.isin(self.spw_array, self.flex_spw_id_array)
-                self.spw_array = self.spw_array[spw_screen]
-                self.Nspws = len(self.spw_array)
-                if self.flex_spw_polarization_array is not None:
-                    self.flex_spw_polarization_array = self.flex_spw_polarization_array[
-                        spw_screen
-                    ]
+            # If nothing to select on, bail!
+            if ind_arr is None:
+                continue
 
-        if pol_inds is not None:
-            self.Npols = len(pol_inds)
-            self.polarization_array = self.polarization_array[pol_inds]
+            for param in self:
+                # For each attribute, if the value is None, then bail, otherwise attempt
+                # to figure out along which axis ind_arr will apply.
+                attr = getattr(self, param)
+                if attr.value is not None:
+                    try:
+                        sel_axis = attr.form.index(key)
+                    except (AttributeError, ValueError):
+                        # If form is not a tuple/list (and therefore not array-like),
+                        # it'll throw an AttributeError, and if key is not found in the
+                        # tuple/list, it'll throw a ValueError. In both cases, skip!
+                        continue
 
-        self.history = self.history + history_update_string
+                    if isinstance(attr.value, np.ndarray):
+                        # If we're working with an ndarray, use take to slice along
+                        # the axis that we want to grab from.
+                        attr.value = attr.value.take(ind_arr, axis=sel_axis)
+                    elif isinstance(attr.value, list):
+                        # If this is a list, it _should_ always have 1-dimension.
+                        assert sel_axis == 0
```

Comment:
Let's add an error message here tells the user this is a bug and they should make an issue.

Reply:
Added in 5279a6f.